packages/typespec-client-generator-core/src/lib.ts

```diff
@@ -159,6 +159,12 @@ export const $lib = createTypeSpecLibrary({
         default: `Cannot pass an empty value to the @clientName decorator`,
       },
     },
+    "duplicate-model-name": {
+      severity: "error",
+      messages: {
+        default: `Duplicate model name: "${"name"}"`,
```

Comment:
Can this also specify the scope, I think otherwise it is going to be confusing.

Reply:
I thought scope is not needed in the first place because it is kind of obvious.
And if we want to add scope to the error message, it will report multiple times for the same duplicates.
But maybe it is more clear to state the scope in the message and throw the same duplicate per scope.

Reply:
added